packages/components/message/src/message.vue

```diff
@@ -55,94 +55,72 @@ import type { BadgeProps } from '@element-plus/components/badge'
+)
+let stopTimer: (() => void) | undefined = undefined
+
+const typeClass = computed(() => {
+  const type = props.type
+  return { [ns.bm('icon', type)]: type && TypeComponentsMap[type] }
+})
 
-  components: {
-    ElBadge,
-    ElIcon,
-    ...TypeComponents,
-  },
-
-  props: messageProps,
-  emits: messageEmits,
-
-  setup(props) {
-    const ns = useNamespace('message')
-    const visible = ref(false)
-    const badgeType = ref<BadgeProps['type']>(
-      props.type ? (props.type === 'error' ? 'danger' : props.type) : 'info'
-    )
-    let stopTimer: (() => void) | undefined = undefined
-
-    const typeClass = computed(() => {
-      const type = props.type
-      return { [ns.bm('icon', type)]: type && TypeComponentsMap[type] }
-    })
-
-    const iconComponent = computed(() => {
-      return props.icon || TypeComponentsMap[props.type] || ''
-    })
-
-    const customStyle = computed<CSSProperties>(() => ({
-      top: `${props.offset}px`,
-      zIndex: props.zIndex,
-    }))
-
-    function startTimer() {
-      if (props.duration > 0) {
-        ;({ stop: stopTimer } = useTimeoutFn(() => {
-          if (visible.value) close()
-        }, props.duration))
-      }
-    }
-
-    function clearTimer() {
-      stopTimer?.()
-    }
-
-    function close() {
-      visible.value = false
-    }
+const iconComponent = computed(() => {
+  return props.icon || TypeComponentsMap[props.type] || ''
+})
 
-    function keydown({ code }: KeyboardEvent) {
-      if (code === EVENT_CODE.esc) {
-        // press esc to close the message
-        if (visible.value) {
-          close()
-        }
-      } else {
-        startTimer() // resume timer
-      }
+const customStyle = computed<CSSProperties>(() => ({
+  top: `${props.offset}px`,
+  zIndex: props.zIndex,
+}))
+
+function startTimer() {
+  if (props.duration > 0) {
+    ;({ stop: stopTimer } = useTimeoutFn(() => {
+      if (visible.value) close()
+    }, props.duration))
+  }
+}
+
+function clearTimer() {
+  stopTimer?.()
+}
+
+function close() {
+  visible.value = false
+}
+
+function keydown({ code }: KeyboardEvent) {
+  if (code === EVENT_CODE.esc) {
+    // press esc to close the message
+    if (visible.value) {
+      close()
     }
+  } else {
+    startTimer() // resume timer
+  }
+}
+
+onMounted(() => {
+  startTimer()
+  visible.value = true
+})
 
-    onMounted(() => {
-      startTimer()
-      visible.value = true
-    })
-
-    watch(
-      () => props.repeatNum,
-      () => {
-        clearTimer()
-        startTimer()
-      }
-    )
-
-    useEventListener(document, 'keydown', keydown)
-
-    return {
-      ns,
-      typeClass,
-      iconComponent,
-      customStyle,
-      visible,
-      badgeType,
+watch(
+  () => props.repeatNum,
+  () => {
+    clearTimer()
+    startTimer()
+  }
+)
 
-      close,
-      clearTimer,
-      startTimer,
-    }
-  },
-})
+useEventListener(document, 'keydown', keydown)
```

Comment:
**Suggestion:** put line#125 inside line#112 `onMounted` body. Although this is not what you made,
but just for correctness. 

Remember to ask me about refactoring a component like https://github.com/element-plus/element-plus/issues/6114#issuecomment-1125552539

Next time, PRs like these will not be accepted if you did not ask in advance.